novawallet/Modules/NetworkDetails/NetworkDetailsInteractor.swift

```diff
@@ -142,18 +165,47 @@ private extension NetworkDetailsInteractor {
                 .item
 
             guard
+                let self,
                 let changedChain,
-                self?.chain != changedChain
+                chain != changedChain
             else {
                 return
             }
 
-            self?.chain = changedChain
-            self?.presenter?.didReceive(updatedChain: changedChain)
+            toggleNodesAfterChainUpdate(
+                beforeChange: chain,
+                updatedChain: changedChain
+            )
+
+            if case let .manual(node) = changedChain.connectionMode {
+                currentSelectedNode = node
+            }
+
+            chain = changedChain
+            presenter?.didReceive(changedChain)
+        }
+
+        chainRegistry.subscribeChainState(self, chainId: chain.chainId)
+    }
+
+    func toggleNodesAfterChainUpdate(
+        beforeChange chain: ChainModel,
+        updatedChain: ChainModel
+    ) {
+        guard chain.syncMode != updatedChain.syncMode else {
+            return
+        }
+
+        if updatedChain.syncMode.enabled() {
+            connectToNodes(of: updatedChain)
+        } else {
+            disconnectNodes()
         }
     }
 
-    func connectToChainNodes() {
+    func connectToNodes(of chain: ChainModel) {
+        nodesConnections = [:]
+
         chain.nodes.forEach { node in
```

Comment:
I would also filter here by wss schema as we might have https nodes for Ethereum to use for one shot connection

Reply:
yep, thanks